src/plugins/common/src/traits/prefab/app.rs

```diff
@@ -0,0 +1,97 @@
+use super::{AddPrefabObserver, Prefab};
+use crate::{errors::Error, systems::log::log};
+use bevy::prelude::*;
+
+impl AddPrefabObserver for App {
+	fn add_prefab_observer<TPrefab, TDependencies>(&mut self) -> &mut Self
+	where
+		TPrefab: Prefab<TDependencies>,
+		TDependencies: 'static,
+	{
+		self.add_observer(instantiate_prefab::<TPrefab, TDependencies>.pipe(log))
+	}
+}
+
+fn instantiate_prefab<TPrefab, TDependencies>(
+	trigger: Trigger<OnInsert, TPrefab>,
```

Comment:
should use `OnAdd` not `OnInsert`. Prefab should only be instantiated once when the component gets added the first time.

Reply:
done